.github/workflows/upgrade-transitive-deps.yml

```diff
@@ -0,0 +1,50 @@
+name: Upgrade transitive dependencies
+
+on:
+  workflow_dispatch: # Allow running on-demand
+  schedule:
+    - cron: '0 3 * * *'
+
+jobs:
+  upgrade:
+    name: Upgrade & Open Pull Request
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: true
+
+      - name: Generate branch name
+        run: |
+          NAME="auto-transitive-dep-upgrades-$(echo "$RANDOM-$RANDOM-$RANDOM" | md5sum | head -c 20)"
```

Comment:
I'd prefer something simpler like `deps/transitive/${{ github.run_id }}`

Reply:
Thanks, I've updated the use this simplified name.